daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -809,11 +829,12 @@ class DirectOrBufferedDataOutputStream private[io] (
   // private def withBitLengthLimit(lengthLimitInBits: Long)(body: => Unit): Boolean = macro IOMacros.withBitLengthLimitMacroForOutput
 
   protected def deliverContent(directDOS: DirectOrBufferedDataOutputStream, finfo: FormatInfo) = {
-
-    val bufOS = this.bufferingJOS
     val bufferNBits = this.relBitPos0b // don't have to subtract a starting offset. It's always zero in buffered case.
+    val bufOS = this.bufferingJOS
 
-    if (directDOS.isEndOnByteBoundary && this.isEndOnByteBoundary) {
+    if (existingFile != null) {
+      directDOS.putFile(bufOS.tempFile.toPath, bufferNBits.toLong, chunkSizeInBytes, finfo)
+    } else if (directDOS.isEndOnByteBoundary && this.isEndOnByteBoundary) {
```

Comment:
It's not clear to my why we need this ``existingFIle != null`` condition. It seems odd this that branch is only taken if existingFIle is set, but not if we switched from buffering to file based? Should those be handled exactly the same? The below code does check if the bufOS.isFile, can't we use that same logic there?